source/Plugins/Process/Dpu/DpuRank.cpp

```diff
@@ -78,10 +56,19 @@ bool DpuRank::Open(llvm::StringRef profile) {
   } else if (args_split.first.equals("simulator")) {
     m_type = FUNCTIONAL_SIMULATOR;
   } else {
-    m_type = GetTypeFromDpuDiagnostic();
+    m_type = HW;
```

Comment:
in the current state of thing, this is mildly satisfying.
Personally I´m often running:
dpu-lldb my_dpu_binary.bin

having HW as default is a long term goal, but we likely need a fallback to simulator ?

Reply:
I mean, we could try to implement something that check if we have any hardware before choosing the default (I am not sure we have everything we need for this in the `hw` backend at the moment). But isn't it why we wanted `UPMEM_PROFILE_BASE` ? To set user specific default properties ?